### ISSUE TITLE
[BUG] menu shows well if gadget attached and adb but not with objection

### ISSUE THREAD

**Issue by VAROME** [commonuserlol/frida-java-menu#5]
**Describe the bug**
i made a full working code , the menu shows fine, changes data fine, on and off working perfectly , toast everything, that is when i connect my phone to the pc with adb and gadget in the apk [Game: Frontier wars] 
the toggle changes the diamond value 
when i use the objection to make a non rooted apk the menu does not show . 

**To Reproduce**
- implement normally a new clean version of frida java version 
`npm i git+https://github.com/commonuserlol/frida-java-menu/
`- build project using esbuild 
`  npx esbuild src/index.ts --bundle --outfile=dist/index.js
`in index.ts code is 
```

import "frida-java-menu";

function main() 
  {
    Menu.toast("this shows so clearly main is called", 4);


    const AssemblyCSharp = Il2Cpp.domain.assembly("Assembly-CSharp").image;
    const set_Diamondclass = AssemblyCSharp.class("GameStatic");
    const set_Diamondmethod = set_Diamondclass.method("get_Diamond");

    let menu = new Menu.JavaMenu("TEST", "Shi1234", Menu.Theme.LGL);
    menu.icon(
      "iVBORw0KGgoAAAANSUhEUgAAACgAAAAiCAIAAACiBwxcAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAALElEQVR42u3NoQ0AAAgDsDkUCf9fyxkzTeqb2auIWCwWi8VisVgsFovF4l7884K/Qa1ytvEAAAAASUVORK5CYII=",
      "Normal"
    );

    menu.add(
      menu.toggle("Button", function (state: boolean) {
        if (state) {
          Menu.toast("checked", 4); 
// diamond changes here .... but the menu not even showing , so no need for this code, if it works it will just toast.
        };
      }
      )
    );

    Menu.toast("this should show but it doesnt", 4);

    menu.show();
  };

 
  Menu.MainActivity.waitForInit(main);

```

- so the toasts all of them show when using gadget and adb
- but when using objection like this it does not: 

config.json 
```

{
  "interaction": {
    "type": "script",
    "path": "libfrida-gadget.script.so",
    "on_load": "resume"
  }
}
```

original apk with only permission to overlay 

`objection patchapk -s apk.apk -c config.json -l index.js --architecture armeabi-v7a
`
NOTE: if i do this with the same diamond changing code without a menu it works fine , the only issue is the menu not showing up when using objection . 

**Expected behavior**
the menu should have shown with objection 

**Smartphone (please complete the following information):**
 - Device non rooted android 11 infinix x695
 - Firmware or Custom ROM: original infinix x695
 - Android version: [Android 11]

**Comment by commonuserlol** [comment]
Alright i will look whats wrong, but one thing is `Menu.toast` which you call with `4`, it accepts only `0` (short, 2s) and `1` (long, 3.5s) due android api

**Comment by VAROME** [comment]
function main() 
is not called when using objection , this is the main issue that we are facing here .


> Alright i will look whats wrong, but one thing is `Menu.toast` which you call with `4`, it accepts only `0` (short, 2s) and `1` (long, 3.5s) due android api

**Comment by VAROME** [comment]
i fixed it

**Comment by commonuserlol** [comment]
Good to know, can you tell a solution (if its directly for this project i can include it) or make pr

**Comment by VAROME** [comment]
ya , the menu wont work if you just add anything related to any other module like il2cpp bridge, so you need to create a workaround this , so i encapsulate the whole menu inside the code of the il2cpp modding itslef not otherwise , so i dont show menu and add a hooking function at the same time , but i make the menu a part of the hooking function itself [encapsulated inside it :D  ] i had to remove everything and tested everything to reach that , its been like 2 days working on it and it is finally complete . 
btw, i invented a code in c++ to hook fields using instances to other classes , but still dd not try on java, if you know one , you could save me the time of creating a whole new one .

**Comment by commonuserlol** [comment]
> so i encapsulate the whole menu inside the code of the il2cpp modding itslef not otherwise
Can you show code or screenshot?

> btw, i invented a code in c++ to hook fields using instances to other classes , but still dd not try on java, if you know one , you could save me the time of creating a whole new one .
What do you mean hook on java? its not possible since java dont have direct access to memory (unlike C/C++)

**Comment by VAROME** [comment]
i do , i hook directly and change what i want , and yes in java , same like i used to do with c++ and it is easy

**Comment by commonuserlol** [comment]
Yes, `waitForInit` required cuz `Java.perform` callback available after java loaded but app code still not executed

> and voila , it works like a charm
so your on/off code will look like
```typescript
function hack1callback(state: boolean) {
 if (state) //run hack
 else //revert
}
function main() {
// ...
    menu.add(menu.toggle("hack 1", hack1callback));
}
```
right?

**Comment by commonuserlol** [comment]
> using LGL mod menu we needed to patch a function like this
> 
> ```
> // GameStatic
> // Token: 0x06000438 RID: 1080 RVA: 0x00002E50 File Offset: 0x00001050
> [Token(Token = "0x60003C1")]
> [Address(RVA = "0x2F77C4", Offset = "0x2F77C4", VA = "0x2F77C4")]
> public static int get_Diamond()
> {
> 	return 0;
> }
> ```
> 
> and we had to use hex values otherwise it wont work , and we even could not hook it to return a custom value instead the known hex , but with frida it is easy done , we hook this . and also this function
> 
> ```
> // GameStatic
> // Token: 0x06000439 RID: 1081 RVA: 0x00002050 File Offset: 0x00000250
> [Token(Token = "0x60003C2")]
> [Address(RVA = "0x2F7850", Offset = "0x2F7850", VA = "0x2F7850")]
> public static void set_Diamond(int value)
> {
> }
> ```
> 
> using lgl or c++ in general you need to create a pointer for this and then hook it , and most of the time you need to use Update offset to make the pointer works but wth frida too , i do the hooking of it same as the get_diamond , no need for pointer , nor update , and not even with offsets , all of this in one function

Yep, C++ hooking its a big pain

**Comment by VAROME** [comment]
no , here is a full code : 
```
import "frida-il2cpp-bridge";
import "frida-java-menu";

// console.log("star ted ...!");

// type Il2CppThis = Il2Cpp.Class | Il2Cpp.Object;

// GameStatic get_Diamond



Il2Cpp.perform(function () {

const AssemblyCSharp = Il2Cpp.domain.assembly("Assembly-CSharp").image;
const set_Diamondclass = AssemblyCSharp.class("GameStatic");
const set_Diamondmethod = set_Diamondclass.method("get_Diamond");




// Il2Cpp.perform(function () { // the shittttttt paaaaaaaiiiii nnnnnnn xxx.xxxx
Java.perform(function () {

  
function main() 
  {

  
    Menu.toast("show this please and work x.x ",0);

    let menu = new Menu.JavaMenu("TEST", "Shi1234", Menu.Theme.LGL);
    Menu.toast("menu defined",0);

    menu.icon(
      "iVBORw0KGgoAAAANSUhEUgAAACgAAAAiCAIAAACiBwxcAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAALElEQVR42u3NoQ0AAAgDsDkUCf9fyxkzTeqb2auIWCwWi8VisVgsFovF4l7884K/Qa1ytvEAAAAASUVORK5CYII=",
      "Normal"
    );
    Menu.toast("icon defined",0);

    menu.add(
      menu.toggle("Button", function (state: boolean) {
       
        
         if (state) {
          Menu.toast("bdamonds hahahhahahahahahha",0);

         
       //    set_Diamondmethod.implementation.... //  when open will put back maybe it gets it killed :idk 





        }
    }
 

     
     
     
      }
      )
    );

    menu.show();
// tooooooooaaaastttt to show , 
  };

 
  main()
  
  // Menu.MainActivity.waitForInit(main); // no more  

});



});
  
```

**Comment by VAROME** [comment]
ya , when i used lgl i choose versioin 3.2 and since then and i create my own c++ codes for hooking , like i created the update instance from antoher class to a class with no update at all lke the weapons of FWD assault , xD

**Comment by commonuserlol** [comment]
So since you fixed this i should close issue

**Comment by VAROME** [comment]
ya

**Comment by commonuserlol** [comment]
Fully fixed in https://github.com/commonuserlol/frida-java-menu/pull/8

**Comment by VAROME** [comment]
> Fully fixed in #8

thanks , will update .
